tools/schematics/workspace-builder/index.ts

```diff
@@ -0,0 +1,103 @@
+// import { chain, externalSchematic, Rule } from '@angular-devkit/schematics';
+
+// import { WorkspaceBuilderSchema } from './schema';
+
+// export default function(schema: WorkspaceBuilderSchema): Rule {
+//   return chain([
+//     externalSchematic('@nrwl/workspace', 'workspace', {
+//       name: schema.name
+//     })
+//   ]);
+// }
+
+import {
+  apply,
+  branchAndMerge,
+  chain,
+  mergeWith,
+  Rule,
+  template,
+  url,
+  move,
+  filter
+} from '@angular-devkit/schematics';
+import {
+  Schema as BuilderSchema,
+  Builder as BuilderDefinition
+} from '@angular-devkit/architect/src/builders-schema';
+import { strings as stringUtils } from '@angular-devkit/core';
+import { formatFiles, toFileName, updateJsonInTree } from '@nrwl/workspace';
+
+import { WorkspaceBuilderSchema as Schema } from './schema';
+
+type NormalizedSchema = Schema & typeof schemaDefaults;
+const schemaDefaults = {
+  autoCompile: true,
+  skipFormat: false
+};
+const workspaceBuildersRoot = 'tools/builders';
+
+export default function(schema: Schema): Rule {
+  const options = normalizeOptions(schema);
+
+  return chain([
+    branchAndMerge(
+      chain([createFiles(options), updateBuildersSchema(options)])
+    ),
+    formatFiles(options)
+  ]);
+}
+
+function createFiles(options: NormalizedSchema): Rule {
+  const templateSource = apply(url('./files'), [
+    template({
```

Comment:
little bit magical for my taste 🧙‍♀️ I would maybe add a comment or two. What do you think?

Reply:
well, it's just interpolating values to template content and template file names

![image](https://user-images.githubusercontent.com/1223799/71835156-28cba200-30b1-11ea-9a6e-7ae0f02d99f5.png)

Reply:
I'll create docs for it (at least to reference library docs)